ZeeHomeScreen/Classes/Components/SectionComposite/SectionCompositeViewController+LazyLoading.swift

```diff
@@ -27,25 +27,34 @@ extension SectionCompositeViewController {
         if let component = currentComponentModel, component.childerns != nil, component.childerns!.count > 0 {
             
             self.liveComponentModel = component
-
-                           if let componentsArray = component.childerns,
-                               componentsArray.count > 0 {
-                               let liveComponentsArray = self.liveComponentsWithLazyLoading(componentsArray: componentsArray, liveComponents: self.sectionsDataSourceArray)
-                               
-                                   if component.isVertical == true && component.type != "GRID" {
-                                       self.prepareCollectionSections(sections: liveComponentsArray)
-                                   }
-                                   else {
-                                       self.prepareCollectionItems(items: liveComponentsArray)
-                                   }
-                           }
-                           else {
-                               // delete lazy loading components if needed
-                           }
-                           
-                           self.setComponentModel(component)
-                           let additionalContent = self.prepareAdditionalContent(component)
-                           self.loadAdditionalContent(indexToInsert: 1, for: additionalContent, component: component)
+            
+            if let componentsArray = component.childerns,
+                componentsArray.count > 0 {
+                let liveComponentsArray = self.liveComponentsWithLazyLoading(componentsArray: componentsArray, liveComponents: self.sectionsDataSourceArray)
+                
+                if component.isVertical == true && !component.isGridType() {
+                    self.prepareCollectionSections(sections: liveComponentsArray)
+                }
+                else {
+                    self.prepareCollectionItems(items: liveComponentsArray)
+                }
+            }
+            else {
+                // delete lazy loading components if needed
+            }
+            
+            self.setComponentModel(component)
+            let additionalContent = self.prepareAdditionalContent(component)
+            self.loadAdditionalContent(indexToInsert: 1, for: additionalContent, component: component)
+            
+            //load next content page if current screen == inner screen
+            if let collectionView = collectionView,
+                component.isVertical == true && component.isGridType() {
+                //there should be a small delay because collectionView crashes when tries to remove loading cell that is not created yet, because response comes immediately (line 303 in this class)
+                DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```

Comment:
Please use a constant for the `0.5` number and use it in both places. Something like `LOADING_EXTRA_CONTENT_DELAY`.

Reply:
done